File: src/test/scala/com/yammer/dropwizard/examples/SplodyCommand.scala
package com.yammer.dropwizard.examples

import com.yammer.dropwizard.cli.{Flag, FlagGroup, Command}
import com.yammer.dropwizard.Service

class SplodyCommand extends Command {
  def name = "splody"

  override def description = Some("Explodes with a Guice error")

  override def options =
    FlagGroup(Seq(Flag("r", "required", "a required option")),required = true) ::
    Flag("e", "exception", "throw an exception") ::
    Flag("m", "message", "return an error message") ::
    Nil

  def run(service: Service, opts: Map[String, List[String]], args: List[String]) = {
    if (opts.contains("exception")) {
      println("Throwing an exception")
      error("EXPERIENCE BIJ")
    }

    opts.get("message").map { _ => "Y U NO DO RIGHT THING" }
  }
}

File: dropwizard/src/test/java/com/yammer/flopwizard/example/SplodyCommand.java
package com.yammer.flopwizard.example;

import com.yammer.dropwizard.Service;
import com.yammer.dropwizard.cli.Command;
import org.apache.commons.cli.CommandLine;
import org.apache.commons.cli.Option;
import org.apache.commons.cli.OptionGroup;
import org.apache.commons.cli.Options;

@SuppressWarnings("NullableProblems")
public class SplodyCommand extends Command {
    public SplodyCommand() {
        super("splody", "throws various errors");
    }

    @Override
    public Options getOptions() {
        final OptionGroup group = new OptionGroup();
        group.setRequired(true);
        group.addOption(new Option("r", "required", false, "a required option"));
        
        return new Options()
                .addOptionGroup(group)
                .addOption("e", "exception", false, "throw an exception");
    }

    @Override
    protected void run(Service service,
                       CommandLine params) throws Exception {
        if (params.hasOption('e')) {
            System.err.println("throwing an exception");
            throw new IllegalAccessError("AW GAWD");
        }
    }
}
